server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -90,20 +89,13 @@
    * @param useTrash
    *          true to move files to trash rather than delete them
    */
-  GarbageCollectWriteAheadLogs(final AccumuloServerContext context, VolumeManager fs,
-      boolean useTrash) throws IOException {
+  GarbageCollectWriteAheadLogs(final AccumuloServerContext context, final VolumeManager fs,
+      final LiveTServerSet liveServers, boolean useTrash) throws IOException {
     this.context = context;
     this.fs = fs;
     this.useTrash = useTrash;
-    this.liveServers = new LiveTServerSet(context, new Listener() {
-      @Override
-      public void update(LiveTServerSet current, Set<TServerInstance> deleted,
-          Set<TServerInstance> added) {
-        log.debug("New tablet servers noticed: " + added);
-        log.debug("Tablet servers removed: " + deleted);
-      }
-    });
-    liveServers.startListeningForTabletServerChanges();
+    this.liveServers = liveServers;
```

Comment:
Could call `liveServers.scanServers()` here making it more functionally equivalent to the old code.  The old code used to call `liveServers.startListeningForTabletServerChanges()` which internally called `scanServers()` and registered a timer to call `scanServers()`.  I don't think registering the timer is needed.

Reply:
One benefit for the timer is that it keeps the tserver set up-to-date so we don't have to wait to scan them in the GC loop. I'm not sure that's a big deal, though. Another is that it lets us know about the changes in the logs.

However, using `scanServers()` directly here would still let us know about changes in the logs... so long as the LiveTServersSet object is kept as a singleton in the SimpleGarbageCollector. The problem with the first "option 2" mentioned in #1314 that used `scanServers()` directly is that a new LiveTServersSet was being created over and over again, so the logs would have always shown an "updated" list of tservers being the entire set of current tservers, before being reset... not very useful. But the `scanServers()` option should work fine now that the LiveTServersSet object is being passed in and reused.

Reply:
I think its best to separate the issue of logging tserver information and reasoning about GC correctness.  The timer seems to add unnecessary concurrency, that while it seems correct does make reasoning about the GC algorithm more complex.

Reply:
The existing code was always calling `scanServers()` because `startListeningForTabletServerChanges()` always called it.

Reply:
The choice is to decide whether `startListeningForTabletServerChanges()` in a single background task every 5 seconds is better, or calling `scanServers()` immediately before using the set is better. I don't know that one is significantly easier to reason about than the other. In both scenarios we are getting a relatively recent view of the current live tservers. But, I do like avoiding the background task if we don't actually need it. SimpleTimer is dubious.